--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -7682,6 +7682,7 @@ static int igb_sriov_reinit(struct pci_d
 	return 0;
 }
 
+#if LINUX_VERSION_IS_GEQ(3,8,0)
 static int igb_pci_disable_sriov(struct pci_dev *dev)
 {
 	int err = igb_disable_sriov(dev);
@@ -7691,6 +7692,7 @@ static int igb_pci_disable_sriov(struct
 
 	return err;
 }
+#endif /* LINUX_VERSION_IS_GEQ(3,8,0) */
 
 static int igb_pci_enable_sriov(struct pci_dev *dev, int num_vfs)
 {
