--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -2078,7 +2078,11 @@ static int igb_set_features(struct net_d
 
 static int igb_ndo_fdb_add(struct ndmsg *ndm, struct nlattr *tb[],
 			   struct net_device *dev,
+#if LINUX_VERSION_IS_LESS(3,19,0)
+			   const unsigned char *addr,
+#else
 			   const unsigned char *addr, u16 vid,
+#endif
 			   u16 flags)
 {
 	/* guarantee we can provide a unique filter for the unicast address */
