diff --git a/compat/lib-refcount.c b/compat/lib-refcount.c
index 5d0582a..8d108f9 100644
--- a/compat/lib-refcount.c
+++ b/compat/lib-refcount.c
@@ -7,6 +7,7 @@
 #include <linux/refcount.h>
 #include <linux/spinlock.h>
 #include <linux/bug.h>
+#include <linux/export.h>
 
 #define REFCOUNT_WARN(str)	WARN_ONCE(1, "refcount_t: " str ".\n")
 
@@ -152,6 +153,8 @@ bool refcount_dec_and_lock(refcount_t *r
 		return false;
 	}
 
+	__release(lock);
+
 	return true;
 }
 EXPORT_SYMBOL(refcount_dec_and_lock);
@@ -181,6 +184,8 @@ bool refcount_dec_and_lock_irqsave(refco
 		return false;
 	}
 
+	__release(lock);
+
 	return true;
 }
 EXPORT_SYMBOL(refcount_dec_and_lock_irqsave);
