--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -158,8 +158,16 @@ static void igb_tx_timeout(struct net_de
 static void igb_reset_task(struct work_struct *);
 static void igb_vlan_mode(struct net_device *netdev,
 			  netdev_features_t features);
+#if LINUX_VERSION_IS_GEQ(3,10,0)
 static int igb_vlan_rx_add_vid(struct net_device *, __be16, u16);
 static int igb_vlan_rx_kill_vid(struct net_device *, __be16, u16);
+#elif LINUX_VERSION_IS_GEQ(3,3,0)
+static int igb_vlan_rx_add_vid(struct net_device *, u16);
+static int igb_vlan_rx_kill_vid(struct net_device *, u16);
+#else
+static void igb_vlan_rx_add_vid(struct net_device *, u16);
+static void igb_vlan_rx_kill_vid(struct net_device *, u16);
+#endif /* LINUX_VERSION_IS_GEQ(3,10,0) */
 static void igb_restore_vlan(struct igb_adapter *);
 static void igb_rar_set_qsel(struct igb_adapter *, u8 *, u32 , u8);
 static void igb_ping_all_vfs(struct igb_adapter *);
@@ -7349,8 +7357,14 @@ static void igb_vlan_mode(struct net_dev
 	igb_set_vf_vlan_strip(adapter, adapter->vfs_allocated_count, enable);
 }
 
+#if LINUX_VERSION_IS_GEQ(3,10,0)
 static int igb_vlan_rx_add_vid(struct net_device *netdev,
 			       __be16 proto, u16 vid)
+#elif LINUX_VERSION_IS_GEQ(3,3,0)
+static int igb_vlan_rx_add_vid(struct net_device *netdev, u16 vid)
+#else
+static void igb_vlan_rx_add_vid(struct net_device *netdev, u16 vid)
+#endif
 {
 	struct igb_adapter *adapter = netdev_priv(netdev);
 	struct e1000_hw *hw = &adapter->hw;
@@ -7362,11 +7376,19 @@ static int igb_vlan_rx_add_vid(struct ne
 
 	set_bit(vid, adapter->active_vlans);
 
+#if LINUX_VERSION_IS_GEQ(3,3,0)
 	return 0;
+#endif /* LINUX_VERSION_IS_GEQ(3,3,0) */
 }
 
+#if LINUX_VERSION_IS_GEQ(3,10,0)
 static int igb_vlan_rx_kill_vid(struct net_device *netdev,
 				__be16 proto, u16 vid)
+#elif LINUX_VERSION_IS_GEQ(3,3,0)
+static int igb_vlan_rx_kill_vid(struct net_device *netdev, u16 vid)
+#else
+static void igb_vlan_rx_kill_vid(struct net_device *netdev, u16 vid)
+#endif /* LINUX_VERSION_IS_GEQ(3,10,0) */
 {
 	struct igb_adapter *adapter = netdev_priv(netdev);
 	int pf_id = adapter->vfs_allocated_count;
@@ -7378,7 +7400,9 @@ static int igb_vlan_rx_kill_vid(struct n
 
 	clear_bit(vid, adapter->active_vlans);
 
+#if LINUX_VERSION_IS_GEQ(3,3,0)
 	return 0;
+#endif /* LINUX_VERSION_IS_GEQ(3,3,0) */
 }
 
 static void igb_restore_vlan(struct igb_adapter *adapter)
@@ -7386,10 +7410,18 @@ static void igb_restore_vlan(struct igb_
 	u16 vid = 1;
 
 	igb_vlan_mode(adapter->netdev, adapter->netdev->features);
+#if LINUX_VERSION_IS_GEQ(3,10,0)
 	igb_vlan_rx_add_vid(adapter->netdev, htons(ETH_P_8021Q), 0);
+#else
+	igb_vlan_rx_add_vid(adapter->netdev, 0);
+#endif /* LINUX_VERSION_IS_GEQ(3,10,0) */
 
 	for_each_set_bit_from(vid, adapter->active_vlans, VLAN_N_VID)
+#if LINUX_VERSION_IS_GEQ(3,10,0)
 		igb_vlan_rx_add_vid(adapter->netdev, htons(ETH_P_8021Q), vid);
+#else
+		igb_vlan_rx_add_vid(adapter->netdev, vid);
+#endif /* LINUX_VERSION_IS_GEQ(3,10,0) */
 }
 
 int igb_set_spd_dplx(struct igb_adapter *adapter, u32 spd, u8 dplx)
