--- a/drivers/nfc/microread/mei.c
+++ b/drivers/nfc/microread/mei.c
@@ -67,7 +67,13 @@ static int microread_mei_remove(struct m
 }
 
 static struct mei_cl_device_id microread_mei_tbl[] = {
+#if LINUX_VERSION_IS_GEQ(4,4,0)
 	{ MICROREAD_DRIVER_NAME, MEI_NFC_UUID, MEI_CL_VERSION_ANY},
+#elif LINUX_VERSION_IS_GEQ(4,2,0)
+	{ MICROREAD_DRIVER_NAME, MEI_NFC_UUID},
+#else
+	{ MICROREAD_DRIVER_NAME},
+#endif
 
 	/* required last entry */
 	{ }
--- a/drivers/nfc/pn544/mei.c
+++ b/drivers/nfc/pn544/mei.c
@@ -67,7 +67,13 @@ static int pn544_mei_remove(struct mei_c
 }
 
 static struct mei_cl_device_id pn544_mei_tbl[] = {
+#if LINUX_VERSION_IS_GEQ(4,4,0)
 	{ PN544_DRIVER_NAME, MEI_NFC_UUID, MEI_CL_VERSION_ANY},
+#elif LINUX_VERSION_IS_GEQ(4,2,0)
+	{ PN544_DRIVER_NAME, MEI_NFC_UUID},
+#else
+	{ PN544_DRIVER_NAME},
+#endif
 
 	/* required last entry */
 	{ }
