--- a/drivers/net/wireless/intersil/prism54/islpci_eth.h
+++ b/drivers/net/wireless/intersil/prism54/islpci_eth.h
@@ -54,6 +54,7 @@ void islpci_eth_cleanup_transmit(islpci_
 netdev_tx_t islpci_eth_transmit(struct sk_buff *, struct net_device *);
 int islpci_eth_receive(islpci_private *);
 void islpci_eth_tx_timeout(struct net_device *, unsigned int txqueue);
+void bp_islpci_eth_tx_timeout(struct net_device *);
 void islpci_do_reset_and_wake(struct work_struct *);
 
 #endif				/* _ISL_GEN_H */
--- a/drivers/net/wireless/intersil/prism54/islpci_eth.c
+++ b/drivers/net/wireless/intersil/prism54/islpci_eth.c
@@ -491,3 +491,10 @@ islpci_eth_tx_timeout(struct net_device
 			"%s: tx_timeout, waiting for reset", ndev->name);
 	}
 }
+
+#if LINUX_VERSION_IS_LESS(5,6,0)
+void bp_islpci_eth_tx_timeout(struct net_device *dev) {
+	islpci_eth_tx_timeout(dev, 0);
+}
+EXPORT_SYMBOL_GPL(bp_islpci_eth_tx_timeout);
+#endif
