--- a/include/net/cfg802154.h
+++ b/include/net/cfg802154.h
@@ -359,6 +359,7 @@ struct wpan_dev {
 
 #define to_phy(_dev)	container_of(_dev, struct wpan_phy, dev)
 
+#if LINUX_VERSION_IS_GEQ(3,19,0)
 static inline int
 wpan_dev_hard_header(struct sk_buff *skb, struct net_device *dev,
 		     const struct ieee802154_addr *daddr,
@@ -369,6 +370,7 @@ wpan_dev_hard_header(struct sk_buff *skb
 
 	return wpan_dev->header_ops->create(skb, dev, daddr, saddr, len);
 }
+#endif /* >= 3.19 */
 
 struct wpan_phy *
 wpan_phy_new(const struct cfg802154_ops *ops, size_t priv_size);
